log/testutil.go

```diff
@@ -34,15 +34,42 @@ func interestingGoroutines() (gs []string) {
 			strings.Contains(stack, "stackimpact-go/internal") ||
 			// stackimpact uses persistent http client conns
 			strings.Contains(stack, "created by net/http.(*Transport).dialConn") ||
-			strings.Contains(stack, "log.MainTest") {
+			strings.Contains(stack, "log.MainTest") ||
+			matchesUserInterestingGoroutine(stack) {
 			continue
 		}
+
 		gs = append(gs, stack)
 	}
 	sort.Strings(gs)
 	return
 }
 
+var userInterestingGoroutines []string
+
+// AddUserInterestingGoroutine can be called when the environment of some
+// specific tests leaks goroutines unknown to interestingGoroutines()
+func AddUserInterestingGoroutine(newGr string) {
+	// Don't add if it's already there
+	for _, gr := range userInterestingGoroutines {
```

Comment:
Why? There's no real harm in duplication, and anyway it's the fault of the caller, so if they want to avoid it they can.